src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1826,16 +1826,17 @@ object Desugar {
     def registerInterface(t: Type.InterfaceT, dT: in.InterfaceT): Unit = {
       Violation.violation(t.decl.embedded.isEmpty, "embeddings in interfaces are currently not supported")
 
-      if (!registeredInterfaces.contains(dT.name)) {
+      if (!registeredInterfaces.contains(dT.name) && info == t.context.getTypeInfo) {
```

Comment:
I assume that this is the core of the fix. In a sense, you only register the interface in the package where it is defined instead of the first desugared package that has a value of the interface type in question right?

Reply:
Yes. Before it was translated for every package where it is used. This would also be fine, but it triggered another bug. That is why I made the other changes too. Now it works without this line, but I added it to avoid redundant computations.